eng/pipelines/common/templates/run-test-job.yml

```diff
@@ -190,6 +191,17 @@ jobs:
         displayName: 'product build'
 
 
+    - ${{ if eq(parameters.runtimeFlavor, 'mono') }}:  
+      # We need to explictly download CoreCLR for Mono
+      - template: /eng/pipelines/common/download-artifact-step.yml
+        parameters:
+          unpackFolder: $(coreClrProductRootFolderPath)
+          artifactFileName: '$(coreClrProductArtifactName)$(archiveExtension)'
+          artifactName: '$(coreClrProductArtifactName)'
+          displayName: 'CoreCLR product download for Mono'
+
+      - script: msbuild $(monoRepoRoot)/mono.proj -target:PatchCoreClrCoreRoot /p:Configuration=Release
```

Comment:
please don't use msbuild directly we try to avoid that. Could you change this to using the build script and -projects argument? This way we ensure we're using the same tools to run stuff.

https://github.com/dotnet/runtime/blob/master/eng/pipelines/common/restore-internal-tools.yml#L7